src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1171,7 +1171,7 @@ private BoundNode RewriteLambdaConversion(BoundLambda node)
                     BoundExpression cache;
                     if (shouldCacheForStaticMethod || shouldCacheInLoop && (object)containerAsFrame != null)
                     {
-                        var cacheVariableType = containerAsFrame?.TypeMap.SubstituteType(type) ?? translatedLambdaContainer;
+                        var cacheVariableType = containerAsFrame?.TypeMap.SubstituteType(node.Type) ?? translatedLambdaContainer;
 
```

Comment:
We are using un-rewritten node.Type here. Other places use rewritten type. Seems very subtle. Could we have a small comment?